### PR TITLE
Fix kqueue-related definitions on *BSD

### PR DESCRIPTION
**Pull request by asomers**
The kqueue-related definitions contained errors for all non-OSX operating
systems.  I split them up into separate sections for macos, freebsd, openbsd,
netbsd and dragonfly, and I removed NOTE_SIGNAL from everything besides OSX
because it's only defined within the kernel.

### CODE REVIEW

File: src/sys/event.rs [hunk 1]
```diff
@@ -96,6 +96,27 @@ pub enum EventFilter {
...
     EVFILT_USER = -9,
+    EVFILT_FS = -10,
+    EVFILT_SYSCOUNT = 10,
+}
+
+#[cfg(target_os = "freebsd")]
+#[repr(i16)] // u_short
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum EventFilter {
+    EVFILT_READ = -1,
```
Comment (posborne) [inline src/sys/event.rs]: Can we use the constants from libc instead (adding as needed).  They have better tests for such things there and it is the way we have been moving things.

**Comment by asomers** [comment]
Yeah, we can do that.  Would you like me to move all of the structure definitions to libc as well?

**Comment by posborne** [comment]
> Yeah, we can do that. Would you like me to move all of the structure definitions to libc as well?

Awesome.  Yeah, I think the structures are probably something that would be good to contribute to libc as well.

**Comment by asomers** [comment]
About to open a new PR that moves most definitions into libc.

**Pull request closed without merging**